src/linpred.jl

```diff
@@ -182,5 +141,24 @@ model_response(obj::LinPredModel) = obj.rr.y
 fitted(m::LinPredModel) = m.rr.mu
 predict(mm::LinPredModel) = fitted(mm)
 formula(obj::LinPredModel) = ModelFrame(obj).formula
-nobs(obj::LinPredModel) = length(model_response(obj))
 residuals(obj::LinPredModel) = residuals(obj.rr)
+
+"""
+    nobs(obj::LinearModel)
+    nobs(obj::GLM)
+
+For linear and generalized linear models, returns the number of rows, or,
+when prior weights are specified, the sum of weights.
+"""
+function nobs(obj::LinPredModel)
+    if isempty(obj.rr.wts)
+        oftype(sum(one(eltype(obj.rr.wts))), length(obj.rr.y))
```

Comment:
I'm not too concerned about booleans, but more about `Int16` and `Int32`. Anyway, weights should be reworked at some point (https://github.com/JuliaStats/GLM.jl/issues/124), and when no weights are present a `Ones` type returning series of `1::Int` would allow use to simplify all of this.